src/corehost/cli/deps_format.cpp

```diff
@@ -43,6 +43,40 @@ void deps_json_t::reconcile_libraries_with_targets(
 
         const auto& properties = library.second.as_object();
 
+        // Get the "path" property, if available.
+        pal::string_t library_path;
+        const auto& library_path_iter = properties.find(_X("path"));
+        if (library_path_iter == properties.end())
+        {
+            library_path = _X("");
+        }
+        else
+        {
+            library_path = library_path_iter->second.as_string();
+
+            if (_X('/') != DIR_SEPARATOR)
+            {
+                replace_char(&library_path, _X('/'), DIR_SEPARATOR);
+            }
+        }
+
+        // Get the "hashPath property, if available.
+        pal::string_t library_hash_path;
+        const auto& library_hash_path_iter = properties.find(_X("hashPath"));
+        if (library_hash_path_iter == properties.end())
+        {
+            library_hash_path = _X("");
+        }
+        else
+        {
+            library_hash_path = library_hash_path_iter->second.as_string();
+
+            if (_X('/') != DIR_SEPARATOR)
+            {
+                replace_char(&library_hash_path, _X('/'), DIR_SEPARATOR);
+            }
+        }
```

Comment:
Can this chunk of code get extracted into a shared function between this and the library code above?  They look identical except for the string property name to look for.